packages/spl-token/src/domain/mint.ts

```diff
@@ -0,0 +1,134 @@
+import { StorageValueStream } from '@aleph-indexer/core'
+import { EventStorage } from '../dal/event.js'
+import {
+  AccountHoldingsFilters,
+  AccountHoldingsOptions,
+  AccountMint,
+  MintEventsFilters,
+} from './types.js'
+import {
+  SPLAccountBalance,
+  SPLAccountHoldings,
+  SPLTokenEvent,
+} from '../types.js'
+import BN from 'bn.js'
+import { Account } from './account.js'
+import { BalanceStateStorage } from '../dal/balanceState.js'
+import { AccountMintStorage } from '../dal/accountMints.js'
+
+export class Mint {
+  constructor(
+    protected address: string,
+    protected eventDAL: EventStorage,
+    protected balanceStateDAL: BalanceStateStorage,
+    protected balanceHistoryDAL: BalanceStateStorage,
+    protected accountMintDAL: AccountMintStorage,
+  ) {}
+
+  async getMintAccounts(
+    account?: string,
+  ): Promise<StorageValueStream<AccountMint>> {
+    const range = account ? [this.address, account] : [this.address]
+    return await this.accountMintDAL.getAllValuesFromTo(range, range)
+  }
+
+  addAccount(account: string): void {
+    const accountMint: AccountMint = {
+      mint: this.address,
+      account,
+    }
+    this.accountMintDAL.save(accountMint)
+  }
+
+  async getEvents(filters: MintEventsFilters): Promise<SPLTokenEvent[]> {
+    let result: SPLTokenEvent[] = []
+    const accountMints = await this.getMintAccounts(filters.account)
+
+    for await (const { account } of accountMints) {
+      const instance = new Account(account, this.eventDAL)
+      const events = await instance.getEvents(filters)
+      result = [...result, ...events]
+    }
+
+    return result
+  }
+
+  async getTokenHolders(): Promise<SPLAccountBalance[]> {
+    const accountMints = await this.getMintAccounts()
+    const balances = []
+
+    for await (const { account } of accountMints) {
+      const balance = await this.balanceStateDAL.getFirstValueFromTo(
```

Comment:
The response here is not sorted by amount, but by account ascii code instead.
Also, I would add the mint field to both, the `account_balance_history` and `account_balance_state` dals, so you can query directly the database in a cleaner way and get the sorted stream.